django/db/models/expressions.py

```diff
@@ -1825,6 +1825,7 @@ class OrderBy(Expression):
     template = "%(expression)s %(ordering)s"
     conditional = False
     constraint_validation_compatible = False
+    allows_composite_expressions = True
```

Comment:
Added in preparation of #18979 and eventual move of `allows_composite_expressions` check to `BaseExpression.resolve_expression`. It's not necessary in the mean time though so this line could be removed and added at a later time.